src/ray/gcs/gcs_server/gcs_job_manager.cc

```diff
@@ -171,6 +171,25 @@ void GcsJobManager::HandleGetAllJobInfo(rpc::GetAllJobInfoRequest request,
     // entrypoint script calls ray.init() multiple times).
     std::unordered_map<std::string, std::vector<int>> job_data_key_to_indices;
 
+    // Create a shared counter for the number of jobs processed
+    std::shared_ptr<std::atomic<int>> num_processed_jobs =
+        std::make_shared<std::atomic<int>>(0);
+
+    // Create a shared boolean flag for the internal KV callback completion
+    std::shared_ptr<std::atomic<bool>> kv_callback_done =
```

Comment:
it doesn't need to be atomic. They are in the same thread.

Reply:
Thanks, fixed